src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,198 @@
+        if (commandParameter.isEmpty()){
+            throw new DukeException("The description of an event task cannot be empty.");
+        }
+
+        String description;
+        String at;
+        try {
+            description = commandParameter.split("/at")[0].trim();
+            at = commandParameter.split("/at")[1].trim();
+        } catch(ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("The at field of the event task is missing.");
+        }
+        if (description.isEmpty()){
+            throw new DukeException("The description of an event task cannot be empty.");
+        }
+
+        tasks.add(new Event(description, at));
+
+        printAddedTask();
+    }
+
+    private static void printAddedTask() {
+        System.out.println("Duke: Got it! I have added this task:");
+        System.out.println(INDENT + tasks.get(tasks.size() - 1).toString());
+        System.out.println(INDENT + "Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println(LINE);
+    }
+
+    private static void listTasks() {
+        System.out.println("Duke: Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println(INDENT + (i + 1) + ". " + tasks.get(i).toString());
+        }
+        System.out.println(LINE);
+    }
+
+    private static String getCommand(String userInput) {
+        return userInput.split(" ")[0];
+    }
+
+    private static String getCommandParameter(String userInput) {
+        String commandParameter = userInput.replace("todo", "")
+                .replace("deadline", "")
+                .replace("event", "")
+                .replace("done", "")
+                .replace("list", "")
+                .trim();
+
+        return commandParameter;
+    }
+
+    private static void markTaskAsDone(String commandParameter) throws DukeException {
+        int taskIndex = getTaskIndex(commandParameter);
+        setTaskAsDone(taskIndex);
+        System.out.println("Duke: I have marked this task as done:");
+        System.out.println(INDENT + tasks.get(taskIndex - 1).toString());
+        System.out.println(LINE);
+    }
+
+    private static void setTaskAsDone(int taskIndex) throws DukeException {
+        try {
+            tasks.get(taskIndex - 1).setDone(true);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("The task index is invalid.");
+        }
+    }
+
+    private static int getTaskIndex(String commandParameter) throws DukeException {
+        if (commandParameter.isEmpty()) {
+            throw new DukeException("The task index is missing.");
+        }
+
+        int taskIndex;
+        try {
+            taskIndex = Integer.parseInt(commandParameter);
+        } catch(NumberFormatException e) {
+            throw new DukeException("The task index is invalid.");
+        }
+
+        return taskIndex;
+    }
+}
```

Comment:
Maybe having the numerous different methods categorised into other classes would be a good idea?